client/app/vps/upgrade/vps-upgrade.controller.js

```diff
@@ -52,10 +55,15 @@ export default class VpsUpgradeCtrl {
    */
   static findMatchingUpgradeOffer(modelType, modelName, modelVersion, availableOffers) {
     // rules are :
+    // - if ref is a special case, map it
     // - if version year is less than 2018, agora version will be 2018v3
     // - otherwise agora version will be 2018v4
     const versionInfos = VpsUpgradeCtrl.parseModelVersion(modelVersion);
-    const destVersion = versionInfos.year < 2018 ? '2018v3' : '2018v4';
+    const destVersion = _.get(
```

Comment:
This is hard to understand :(

Maybe make a dedicated function ?

Reply:
With the explanations above in comment difficult to explain more than that :-$

Reply:
I can't read the comment, Github doesn't want me to 😅